pkg/controller/nodeipam/ipam/range_allocator.go

```diff
@@ -295,7 +302,15 @@ func (r *rangeAllocator) filterOutServiceRange(serviceCIDR *net.IPNet) {
 	// clusterCIDR's Mask applied (this means that clusterCIDR contains
 	// serviceCIDR) or vice versa (which means that serviceCIDR contains
 	// clusterCIDR).
+	serviceCIDRv6 := netutils.IsIPv6CIDR(serviceCIDR)
 	for idx, cidr := range r.clusterCIDRs {
+		// if they are not of the same IPFamily then ignore the filtering
+		clusterCIDRv6 := netutils.IsIPv6CIDR(cidr)
```

Comment:
do we need to do this or does `Contains` do the right thing?

https://play.golang.org/p/IF80enFQATU

Reply:
yes. I just checked the source 
```
// Contains reports whether the network includes ip.
func (n *IPNet) Contains(ip IP) bool {
	nn, m := networkNumberAndMask(n)
	if x := ip.To4(); x != nil {
		ip = x
	}
	l := len(ip)
	if l != len(nn) {
		return false
	}
	for i := 0; i < l; i++ {
		if nn[i]&m[i] != ip[i]&m[i] {
			return false
		}
	}
	return true
}
```

There is a `len` comparison. removing the family-match check.